app/controllers/application_controller.rb

```diff
@@ -1,4 +1,28 @@
 class ApplicationController < ActionController::Base
   # Only allow modern browsers supporting webp images, web push, badges, import maps, CSS nesting, and CSS :has.
   allow_browser versions: :modern
+
+  before_action :require_login
+
+  # Finds the User with the ID stored in the session with the key :current_user_id.
+  def current_user
+    @_current_user ||= session[:current_user_id] &&
```

Comment:
alguma razão para esta var de instância ter um underscore antes do nome?

Reply:
Segui a [documentação](https://guides.rubyonrails.org/action_controller_overview.html#accessing-the-session). Também me perguntei pelo mesmo mas não encontrei resposta. Presumo que seja para atribuir um significado especial à variável. Posso remover :)

Reply:
Aparentemente é uma convenção para sinalizar que a variável deve ser apenas usada no controlador, por oposição a variáveis de instância acedidas por views.

Reply:
Acabo por usar a variável no ApplicationController e no LoginsController mas no PR #13 criei os métodos `login` e `logout` no ApplicationController, que são depois invocados nos mesmos sítios.